torch/csrc/distributed/c10d/TraceUtils.h

```diff
@@ -699,7 +704,9 @@ struct NCCLTraceBuffer {
       };
 
       dict.insert(input_sizes_key, read_sizes(e.input_dims_));
+      dict.insert(input_dtypes_key, e.input_dtypes_);
```

Comment:
one thing i wonder is, could we have statically allocated string objects that represent each of the dtype integers, and then insert them into the dicts at dump time rather than inserting the raw int?

or, at least, write a separate section into the dict which is a dtype LUT from which the later tooling/script can replace '6' with the readable value?

Reply:
I changed the code to simply store and print out the strings.
Example record with change is below:

```
{'record_id': 1, 'pg_id': 0, 'process_group': ('0', 'default_pg'), 'collective_seq_id': 2, 'p2p_seq_id': 0, 'op_id': 2, 'profiling_name': 'nccl:all_reduce', 'time_created_ns': 1716252027840457034, 'input_sizes': [[3, 4]], 'input_dtypes': ['float'], 'output_sizes': [[3, 4]], 'output_dtypes': ['float'], 'state': 'completed', 'time_discovered_started_ns': None, 'time_discovered_completed_ns': 1716252027858706135, 'retired': True, 

```